src/main/kotlin/com/hibob/academy/dao/models.kt

```diff
@@ -19,7 +19,19 @@ data class PetCreationRequest(
     val companyId: Long,
     val arrivalDate: LocalDate,
     val ownerId: Long?
-)
+) {
+    fun extractPetCreationToPet(petCreationRequest: PetCreationRequest, id: Long): Pet {
```

Comment:
rename to fromPetCreation since you are inside of a Pet data class already.
Second, the id should be the first parameter.

Reply:
You right, Im changing it